test/functional/api/buffer_spec.lua

```diff
@@ -423,6 +423,13 @@ describe('api/buf', function()
       -- will join multiple lines if needed
       set_text(0, 6, 3, 4, {'bar'})
       eq({'hello bar'}, get_lines(0,  1, true))
+
+      -- can use negative line numbers
+      set_text(-2, 0, -2, 5, {'goodbye'})
+      eq({'goodbye bar'}, get_lines(0, 1, true))
```

Comment:
I'd suggest using `get_lines(0, -1, true)` as to see the complete result on the buffer